subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoTaskExtension.java

```diff
@@ -263,13 +263,19 @@ public void setJmx(boolean jmx) {
     /**
      * Gets all properties in the format expected of the agent JVM argument.
      *
+     * @param useAbsolutePaths return all paths as absolute paths
+     *
      * @return state of extension in a JVM argument
      */
-    public String getAsJvmArg() {
+    public String getAsJvmArg(boolean useAbsolutePaths) {
         StringBuilder builder = new StringBuilder();
-        ArgumentAppender argument = new ArgumentAppender(builder, task.getWorkingDir());
+        ArgumentAppender argument = new ArgumentAppender(builder, task.getWorkingDir(), useAbsolutePaths);
         builder.append("-javaagent:");
-        builder.append(GFileUtils.relativePath(task.getWorkingDir(), agent.getJar()));
+        if (useAbsolutePaths) {
+            builder.append(agent.getJar().getAbsoluteFile());
```

Comment:
FWIW Wondering why path to agent should be optional/configurable? Is it because of #949 ? In any case IMO configurability here is not very convenient and not very intuitive.